### PR TITLE
Org models

### LINKED ISSUE #30: Create models of org app
**Is your feature request related to a problem? Please describe.**
This issue should be attempted only after #29. Create models of org app. The db design is provided here https://dbdiagram.io/d/5e97237739d18f5553fda63b.

**Describe the solution you'd like**
Create these models in the org app:
1. Org
1. Group
1. Member
1. PermissionSet
1. Leaderboard

Generate a migrations file after creating the model.

Make use of PermissionsSetField Created in the #29 in PermissionSet model.

### PR DESCRIPTION
**Pull request by priyanshisharma**
### Description
Created and migrated mentioned models with reference to the dbdiagram.

**Type of Change:** (Delete irrelevant options)
- Code

**Code/Quality Assurance Only** (Delete irrelevant options)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test.

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance**
- [x] My changes generate no new warnings

### CODE REVIEW

File: org/models.py [hunk 1]
```diff
@@ -1,10 +1,57 @@
 from django.db import models
+from django.utils import timezone
+from .custom_model_field import *
+from users.models import User
 
 # Create your models here.
-from .custom_model_field import *
 
-class PermissionSetTestModel(models.Model):
-    perm = PermissionField()
```
Comment (naveennvrgup) [inline org/models.py]: Please dont remove this model as any future changes in the PermissionField model field is needed to be tested. And the testing of migrations is done with the help of this model.

File: org/models.py [hunk 2]
```diff
@@ -1,10 +1,57 @@
...
+class Org (models.Model):
+    route_slug = models.SlugField(max_length = 40)
```
Comment (naveennvrgup) [inline org/models.py]:
```suggestion
    route_slug = models.SlugField(max_length = 40, unique=True)
```

File: org/models.py [hunk 3]
```diff
@@ -1,10 +1,57 @@
...
+    can_join_without_invite = models.BooleanField(default = True)
+    name = models.CharField(max_length=30)
+    tagline = models.CharField(max_length = 50)
+    about = models.CharField(max_length = 500)
```
Comment (naveennvrgup) [inline org/models.py]:
```suggestion
    about = models.CharField(max_length = 500, null=True)
```

File: org/models.py [hunk 4]
```diff
@@ -1,10 +1,57 @@
...
+    profile_pic = models.ImageField(upload_to = 'teesco-backend/static/profile_pic')
+    cover_pic = models.ImageField(upload_to = 'teesco-backend/static/cover_pic')
+    created_at = models.DateTimeField(default = timezone.now)
```
Comment (naveennvrgup) [inline org/models.py]:
```suggestion
    profile_pic = models.ImageField(upload_to = 'uploads/org/profile_pic', null=True)
    cover_pic = models.ImageField(upload_to = 'uploads/org/cover_pic', null=True)
    created_at = models.DateTimeField(auto_now_add=True)
```

File: org/models.py [hunk 5]
```diff
@@ -1,10 +1,57 @@
...
+
+    def __str__(self):
+        return self.name
+
+class PermissionSet(models.Model):
+    org = models.ForeignKey(Org,on_delete=models.CASCADE)
+    permissions = PermissionField()
+
```
Comment (naveennvrgup) [inline org/models.py:27]:
```suggestion
     name = models.CharField(max_length=30)
```

File: org/models.py [hunk 6]
```diff
@@ -1,10 +1,57 @@
...
+    def __str__(self):
+        return f'{self.id}-{self.perm.permissions_to_integer()}'
+
+
+
+class Group (models.Model):
+    name = models.CharField(max_length = 30)
+    role = models.CharField(max_length = 200)
+    invite_slug = models.SlugField(max_length = 40)
```
Comment (naveennvrgup) [inline org/models.py]:
```suggestion
    invite_slug = models.SlugField(max_length = 40, unique=True)
```

File: org/models.py [hunk 7]
```diff
@@ -1,10 +1,57 @@
...
+
+
+class Member (models.Model):
+    user = models.ForeignKey(User,on_delete=models.CASCADE)
+    org = models.ForeignKey(Org,on_delete=models.CASCADE)
+    group  = models.ForeignKey(Group,on_delete=models.CASCADE)
+    permissions  = models.ForeignKey(PermissionSet, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return self.user.name + self.id
```
Comment (naveennvrgup) [inline org/models.py]: __str__() can return < id > - < org.name > - < group.name > - < user.name >

File: org/models.py [hunk 8]
```diff
@@ -1,10 +1,57 @@
...
+
+class Leaderboard (models.Model):
+    user = models.ForeignKey(User,on_delete=models.CASCADE)
+    points = models.IntegerField()
+    org = models.ForeignKey(Org,on_delete=models.CASCADE)
 
     def __str__(self):
-        return f'{self.id}-{self.perm.permissions_to_integer()}'
\ No newline at end of file
+        return self.user.name + self.id
```
Comment (naveennvrgup) [inline org/models.py]: __str__() can return < id > - < org.name > - < user.name > - < points >

**Review by naveennvrgup — COMMENTED** [review]
Great job on the PR. Please added my comments :fox_face:

**Comment by priyanshisharma** [comment]
I've committed the branch with the recommended changes. Please have a look.

**Comment by priyanshisharma** [comment]
Thank you sir! I've committed the branch with the recommended changes. Please have a look.

File: org/models.py [hunk 9]
```diff
@@ -1,10 +1,60 @@
...
 # Create your models here.
-from .custom_model_field import *
 
 class PermissionSetTestModel(models.Model):
     perm = PermissionField()
 
+class Org (models.Model):
+    route_slug = models.SlugField(max_length = 40, unique=True)
+    can_join_without_invite = models.BooleanField(default = True)
+    name = models.CharField(max_length=30)
```
Comment (sumit-badsara) [inline org/models.py]:
```suggestion
    name = models.CharField(max_length=30, null=False, blank=False)
```

File: org/models.py [hunk 10]
```diff
@@ -1,10 +1,60 @@
...
+    def __str__(self):
+        return self.name
+
+class PermissionSet(models.Model):
+    name = models.CharField(max_length=30)
+    org = models.ForeignKey(Org,on_delete=models.CASCADE)
+    permissions = PermissionField()
+
+    def __str__(self):
+        return f'{self.id}-{self.perm.permissions_to_integer()}'
```
Comment (sumit-badsara) [inline org/models.py]: `self.perm` isn't defined, use `self.permissions`

File: org/models.py [hunk 11]
```diff
@@ -1,10 +1,60 @@
...
+
+
+
+class Group (models.Model):
+    name = models.CharField(max_length = 30)
```
Comment (sumit-badsara) [inline org/models.py]:
```suggestion
    name = models.CharField(max_length = 30, null=False, blank=False)
```

File: org/models.py [hunk 12]
```diff
@@ -1,10 +1,60 @@
...
+
+
+class Member (models.Model):
+    user = models.ForeignKey(User,on_delete=models.CASCADE)
+    org = models.ForeignKey(Org,on_delete=models.CASCADE)
+    group  = models.ForeignKey(Group,on_delete=models.CASCADE)
+    permissions  = models.ForeignKey(PermissionSet, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return f'{self.id}-{self.org.name}-{self.group.name}-{self.user.name}'
```
Comment (sumit-badsara) [inline org/models.py]: maybe `self.user.username` instead of `self.user.name`?

**Review by sumit-badsara — COMMENTED** [review]
Almost done I guess :octocat: ! Great Work :fire:  

Few more suggestions :
1. name should be required for all the models
2. username of user should be used instead of name (Is it even added to user model? Not sure)

File: org/models.py (hunk 12 above)
Comment (priyanshisharma) [reply org/models.py:51]: In the user model, there is no username define, shall I add a username to it?

**Comment by priyanshisharma** [comment]
Thanks sir! I've made the specified changes in the org models, and have also added the username field to the user model and ran migrations. Please have a look.

**Review by naveennvrgup — APPROVED** [review]
1. Tested locally. Works great. 
1. Please dont resolve the conversations of your own unless it is a compliment.
1. It has to be resolved by the person who started it.
1. Merging the PR.

**Pull request merged**